dev/dev-env-setup.sh

```diff
@@ -0,0 +1,158 @@
+    q) quiet=1 ;;
+    h) showHelp; exit ;;
+    *) showHelp; exit ;;
+  esac
+done
+
+if [[ $verbose = 1 ]]; then
+  set -exv
+fi
+
+# Check if pyenv is installed and offer to install it if not present
+pyenv_exist=$(command -v pyenv)
+
+if [ -z "$pyenv_exist" ]; then
+  read -p "pyenv is required to be installed to manage python versions. Would you like to install it? $(tput bold)(y/n)$(tput sgr0): " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    echo "Updating brew and installing pyenv..."
+    echo "Note: this will probably take a considerable amount of time."
+    brew update
+    brew install pyenv
+  else
+    exit 1
+  fi
+fi
+
+MLFLOW_HOME=$(pwd)
+
+# Get the minimum supported version from MLflow to ensure any feature development adheres to legacy Python versions
+min_py_version=$(grep "python_requires=" "$MLFLOW_HOME/setup.py" | grep -E -o "([0-9]{1,}\.)+[0-9]{1,}")
+
+# Resolve a minor version to the latest micro version
+case $min_py_version in
+  "3.7") PY_INSTALL_VERSION="3.7.13" ;;
+  "3.8") PY_INSTALL_VERSION="3.8.13" ;;
+  "3.9") PY_INSTALL_VERSION="3.9.11" ;;
+  "3.10") PY_INSTALL_VERSION="3.10.3" ;;
+esac
+
+# Install the Python version if it cannot be found
+pyenv install -s $PY_INSTALL_VERSION
+pyenv local $PY_INSTALL_VERSION
+pyenv exec pip install "$( (( $quiet == 1 && $verbose == 0 )) && printf %s '-q' )" --upgrade pip
+pyenv exec pip install "$( (( $quiet == 1 && $verbose == 0 )) && printf %s '-q' )" virtualenv
+
+VENV_DIR="$directory/bin/activate"
+
+# Check if the virtualenv already exists at the specified path
+if [[ -d "$directory" ]]; then
+  read -p "A virtual environment is already located at $VENV_DIR. Do you wish to replace it? $(tput bold; tput setaf 2)(y/n) $(tput sgr0)" -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    deactivate
+    rm -rf "$directory"
+    echo "Virtual environment removed from '$directory'. Installing new instance."
+    pyenv exec virtualenv "$directory"
+  fi
+else
+  # Create a virtual environment with the specified Python version
+  pyenv exec virtualenv "$directory"
+fi
+
+# Activate the virtual environment
+# shellcheck source=/dev/null
+source "$VENV_DIR"
+
+echo "$(tput setaf 2)Current Python version: $(tput bold)$(python --version)$(tput sgr0)"
+echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/bin/activate$(tput sgr0)"
+
+echo "Installing pip dependencies for development environment."
+# Install required dependencies for Prophet
+tmp_dir=$(mktemp -d)
+pip download "$( (( $quiet == 1 && $verbose == 0 )) && printf %s '-q' )" --no-deps --dest "$tmp_dir" --no-cache-dir prophet
+tar -zxvf "$tmp_dir"/*.tar.gz -C "$tmp_dir"
+pip install "$( (( $quiet == 1 && $verbose == 0 )) && printf %s '-q' )" -r "$(find "$tmp_dir" -name requirements.txt)"
+rm -rf "$tmp_dir"
+
+# Install dev requirements and test plugin
+pip install "$( (( $quiet == 1 && $verbose == 0 )) && printf %s '-q' )" -r "$MLFLOW_HOME/requirements/dev-requirements.txt"
```

Comment:
Core maintainers like us need all the extra dependencies, but most contributors don't, and installing all of them is difficult to succeed. Can we install a minimum set of dev dependencies by default?

Reply:
Should we make the [extras] optional? Or:

Options:
--packages 
"light" - skinny, small, lint
"full" - light + large, doc, extra-ml
"all" - dev-requirements + [extras]

WDYT?